src/commands/infra/init.ts

```diff
@@ -0,0 +1,94 @@
+import child_process from "child_process";
+import commander from "commander";
+import emoji from "node-emoji";
+import { promisify } from "util";
+import { logger } from "../../logger";
+
+/**
+ * Adds the init command to the commander command object
+ *
+ * @param command Commander command object to decorate
+ */
+
+// Check for versions for each executable
+
+let binaries: string[] = ["terraform", "git", "az", "helm"];
+let envVar: string[] = [
+  "ARM_SUBSCRIPTION_ID",
+  "ARM_CLIENT_ID",
+  "ARM_CLIENT_SECRET",
+  "ARM_TENANT_ID"
+];
+
+export const initCommand = (command: commander.Command): void => {
+  command
+    .command("init")
+    .alias("i")
+    .description(
+      "Initialize will verify that all infrastructure deployment prerequisites have been correctly installed."
+    )
+    .action(async opts => {
+      try {
+        if (await validatePrereqs(binaries)) {
```

Comment:
Instead of nesting can put this all in one conditional? Also we don't use the bool return value from `validateEnvVariables(envVar)`

Reply:
We are using nested if statements to allow the CLI to terminate if it errors out. So we only allow the CLI to proceed to the next function if it passes the previous one. We still are relying on a returned boolean in order to unit test `validateEnvVariables()`

Reply:
I understand. If we are modifying our prod code to accommodate tests in that manner it could indicate we can test in another way. Just a thought.

Reply:
That's fair, and will definitely keep this mind. I have created a bool return with the intention that the `init` command will extend with additional functions.